gradle tooling not JDK 20 compatible

### Describe the bug

We run integration tests (e.g. `io.quarkus.gradle.nativeimage.BasicJavaNativeBuildIT`) on latest JDK 20 early access builds and we see a failure in the gradle tooling. It looks like gradle doesn't like the bytecode version just yet:

```
 Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 64
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:180)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:287)
	... 176 more
```

See: https://github.com/graalvm/mandrel/actions/runs/4140128711/jobs/7158708230#step:11:198

### Expected behavior

Gradle tooling works with JDK 20

### Actual behavior

Fails to eat JDK 20 produced bytecode.

### How to Reproduce?

Run the mentioned integration test on JDK 20 EA.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -64,7 +64,7 @@
         <!-- These 2 properties are used by CreateProjectMojo to add the Maven Wrapper -->
         <proposed-maven-version>3.9.3</proposed-maven-version>
         <maven-wrapper.version>3.2.0</maven-wrapper.version>
-        <gradle-wrapper.version>8.1.1</gradle-wrapper.version>
+        <gradle-wrapper.version>8.3</gradle-wrapper.version>
         <quarkus-gradle-plugin.version>${project.version}</quarkus-gradle-plugin.version>
         <quarkus-maven-plugin.version>${project.version}</quarkus-maven-plugin.version>
         <maven-plugin-plugin.version>3.8.1</maven-plugin-plugin.version>

--- a/devtools/gradle/gradle/wrapper/gradle-wrapper.properties
+++ b/devtools/gradle/gradle/wrapper/gradle-wrapper.properties
@@ -1,8 +1,8 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 # https://gradle.org/release-checksums/
-distributionSha256Sum=5625a0ae20fe000d9225d000b36909c7a0e0e8dda61c19b12da769add847c975
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-all.zip
+distributionSha256Sum=bb09982fdf52718e4c7b25023d10df6d35a5fff969860bdf5a5bd27a3ab27a9e
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-all.zip
 networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -77,7 +77,7 @@
         <plexus-utils.version>3.5.1</plexus-utils.version>
         <smallrye-common.version>2.1.0</smallrye-common.version>
         <smallrye-beanbag.version>1.3.2</smallrye-beanbag.version>
-        <gradle-tooling.version>8.1.1</gradle-tooling.version>
+        <gradle-tooling.version>8.3</gradle-tooling.version>
         <quarkus-fs-util.version>0.0.9</quarkus-fs-util.version>
         <org-crac.version>0.1.3</org-crac.version>
         <formatter-maven-plugin.version>2.23.0</formatter-maven-plugin.version>

--- a/independent-projects/tools/devtools-testing/src/main/resources/fake-catalog.json
+++ b/independent-projects/tools/devtools-testing/src/main/resources/fake-catalog.json
@@ -388,7 +388,7 @@
         "supported-maven-versions": "[3.6.2,)",
         "proposed-maven-version": "3.8.8",
         "maven-wrapper-version": "3.2.0",
-        "gradle-wrapper-version": "8.1.1"
+        "gradle-wrapper-version": "8.3"
       }
     },
     "codestarts-artifacts": [

--- a/independent-projects/tools/pom.xml
+++ b/independent-projects/tools/pom.xml
@@ -42,7 +42,7 @@
         <!-- These properties are used by CreateProjectMojo to add the Maven Wrapper -->
         <proposed-maven-version>3.9.3</proposed-maven-version>
         <maven-wrapper.version>3.2.0</maven-wrapper.version>
-        <gradle-wrapper.version>8.1.1</gradle-wrapper.version>
+        <gradle-wrapper.version>8.3</gradle-wrapper.version>
 
         <!-- These properties are needed in order for them to be resolvable by the generated projects -->
         <compiler-plugin.version>3.11.0</compiler-plugin.version>

--- a/integration-tests/gradle/gradle/wrapper/gradle-wrapper.properties
+++ b/integration-tests/gradle/gradle/wrapper/gradle-wrapper.properties
@@ -1,8 +1,8 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 # https://gradle.org/release-checksums/
-distributionSha256Sum=5625a0ae20fe000d9225d000b36909c7a0e0e8dda61c19b12da769add847c975
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-all.zip
+distributionSha256Sum=bb09982fdf52718e4c7b25023d10df6d35a5fff969860bdf5a5bd27a3ab27a9e
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-all.zip
 networkTimeout=10000
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

